plugin/cog/cog-reader/src/main/java/it/geosolutions/imageioimpl/plugins/cog/CogImageReaderSpi.java

```diff
@@ -78,4 +78,14 @@ public boolean canDecodeInput(Object input) throws IOException {
         );
     }
 
+    @Override
+    public void onRegistration(ServiceRegistry registry, Class category) {
```

Comment:
@aaime sure. 
The CogImageReaderSpi extends TIFFImageReaderSpi.
The onRegistration method of the TIFFImageReaderSpi will deregister other TIFF related Spis.

Therefore the CogImageReaderSpi (extending the TIFFImageReaderSpi) was actually resulting in deregistering other TIFF plugins (so the parent class TIFFImageReaderSpi).
The above override on the COG SPI does nothing so that both CogImageReaderSpi and TIFFImageReaderSpi will survive to the registration process.

Reply:
Good to go IMHO.